app/components/memorable_date_component.rb

```diff
@@ -0,0 +1,154 @@
+##
+# Provides a component that accepts a date using the inputs specified
+# by USWDS here: https://designsystem.digital.gov/components/memorable-date/
+#
+# This treats the month, day, and year as nested fields to allow for various
+# uses of the component. Translate the fields in the controller (or equivalent)
+# if you need them to represent a single value.
+#
+# The errors for this component are configurable via +error_messages+, and you may
+# include custom min/max validations via +range_errors+.
+class MemorableDateComponent < BaseComponent
+  attr_reader :name, :month, :day, :year, :required, :hint, :label, :form
+
+  alias_method :f, :form
+
+  ##
+  # @param [String] name Field name for hash containing month, day, and year
+  # @param [String] hint Additional Hint to show to the user
+  # @param [String] label Label for field
+  # @param [String] form Form that this field belongs to
+  # @param [String] month Starting value for month
+  # @param [String] day Starting value for day
+  # @param [String] year Starting value for year
+  # @param [String] required Whether this field is required
+  # @param [Date,DateTime] min Minimum allowed date, inclusive
+  # @param [Date,DateTime] max Maximum allowed date, inclusive
```

Comment:
```suggestion
  # @param [Date,#to_date] min Minimum allowed date, inclusive
  # @param [Date,#to_date] max Maximum allowed date, inclusive
```
this makes it "anything that is a date, or could easily be converted to one"